### PR TITLE
Feature/subpackage merge

### PR DESCRIPTION
**Pull request by StokesMIDE**
Combines all the subpackages (and their tests, etc.) into the 'main' package. Recent changes to the subpackages will need to be merged in later.

### CODE REVIEW

**Comment by CFlaniganMide** [comment]
Weren't we intending to wait on adding Batch in?

**Comment by CrepeGoat** [comment]
> Weren't we intending to wait on adding Batch in?

that was my impression as well

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - run: python -m pip install --upgrade pip
+
+      - uses: actions/checkout@v2
+
+      - run: python -m pip install .[test]
+
+      - run: python -m pytest tests --cov=endaq.batch
```
Comment (CrepeGoat) [inline .github/workflows/unit-tests.yml:34]: small fix:
```suggestion
      - run: python -m pytest tests --cov=endaq
```

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+pandas>=1.3
+plotly>=5.3.1
+python-dotenv>=0.18.0
+requests>=2.25.1
+scipy>=1.7.1
+
+# Testing
+hypothesis
+numpy-quaternion==2020.11.2.17.0.49
+plotly
```
Comment (CrepeGoat) [inline requirements.txt:16]: remove unnecessary test requirements (due to removing `endaq-batch`)?
```suggestion
```

File: setup.py [hunk 3]
```diff
@@ -4,13 +4,25 @@
...
+    "plotly>=5.3.1",
+    "python-dotenv>=0.18.0",
+    "requests>=2.25.1",
+    "scipy>=1.7.1",
     ]
 
 TEST_REQUIRES = [
+    "hypothesis",
+    "numpy-quaternion==2020.11.2.17.0.49",
+    "plotly",
```
Comment (CrepeGoat) [inline setup.py:22]: remove unnecessary test requirements (due to removing `endaq-batch`)?
```suggestion
```

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
+pandas>=1.1.5
+pandas>=1.3
+plotly>=5.3.1
+python-dotenv>=0.18.0
+requests>=2.25.1
+scipy>=1.7.1
+
+# Testing
+hypothesis
+plotly
```
Comment (CrepeGoat) [inline requirements.txt:15]: this requirement is already listed on line 8
```suggestion
```

File: setup.py [hunk 5]
```diff
@@ -4,13 +4,24 @@
...
+    "pandas>=1.3",
+    "plotly>=5.3.1",
+    "python-dotenv>=0.18.0",
+    "requests>=2.25.1",
+    "scipy>=1.7.1",
     ]
 
 TEST_REQUIRES = [
+    "hypothesis",
+    "plotly",
```
Comment (CrepeGoat) [inline setup.py:21]: this requirement is already listed on line 13
```suggestion
```

**Review by CrepeGoat — CHANGES_REQUESTED** [review]

File: setup.py (hunk 5 above)
Comment (StokesMIDE) [reply setup.py:21]: `endaq.plot` appears to use it.

Comment (StokesMIDE) [reply setup.py:21]: Previous comment was regarding its removal along with `numpy-quaternion`.

Comment (CrepeGoat) [reply setup.py:21]: > `endaq.plot` appears to use it.

yes that's perfectly fine, but the `setup.py` file and the `requirements.txt` files both list `plotly` as a requirement **TWICE**. what I'm asking is to please remove the second instances from those two files.

Comment (CrepeGoat) [reply setup.py]: thanks 👍

File: requirements.txt (hunk 4 above)
Comment (CrepeGoat) [reply requirements.txt]: addressed in 8461b0c63347c7607bc29b54840af15e79d51bb1

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (CrepeGoat) [reply .github/workflows/unit-tests.yml]: addressed in a7ad2fe97eecb081fd0361ea02360240d15713fc

File: requirements.txt (hunk 2 above)
Comment (CrepeGoat) [reply requirements.txt]: addressed in a7ad2fe97eecb081fd0361ea02360240d15713fc & 8461b0c63347c7607bc29b54840af15e79d51bb1

File: setup.py (hunk 3 above)
Comment (CrepeGoat) [reply setup.py]: addressed in a7ad2fe97eecb081fd0361ea02360240d15713fc & 8461b0c63347c7607bc29b54840af15e79d51bb1

**Comment by CFlaniganMide** [comment]
@StokesMIDE what's left before this is ready for review?

**Comment by StokesMIDE** [comment]
> @StokesMIDE what's left before this is ready for review?

Since all the tests are in, I'm going to change this to a 'real' PR.

**Review by CFlaniganMide — APPROVED** [review]
This all looks good.

File: setup.py [hunk 6]
```diff
@@ -23,7 +33,7 @@
 
 setuptools.setup(
```
Comment (CrepeGoat) [inline setup.py:34]: do we want to change `classifiers` to remove Python 3.5 & 3.6?

**Review by CrepeGoat — APPROVED** [review]
one inconsequential in-line question, but it seems good.

On a side-note, there's a couple bits from plot that's not in this PR (that's understandable & easy to fix later), but I want to list out specifics for the record:
- MideTechnology/endaq-python-plot#30
- MideTechnology/endaq-python-plot#31

**Pull request merged**